File: test/data/cs/test02/StateState2.cs
public class StateState2 : BaseState
{
    private StateController _stateController;
    private IControllee _controllee;
    public StateState2(StateController stateController, IControllee controllee) : base(controllee)
    {
        _stateController = stateController;
        _controllee = controllee;
    }
    public override BaseState? TryTransitWithoutEvent()
    {
        _controllee.DoAction3();
        return null;
    }
    public override string GetStateName()
    {
        return "State2";
    }
    public override BaseState? GetParent()
    {
        return null;
    }
    public override int GetStateID()
    {
        return 1;
    }
}

File: test/data/cpp/test02/StateState2.cpp
#include "StateState2.h"
StateState2::StateState2(StateController* stateController, IControllee* controllee) : BaseState(controllee)
{
    _stateController = stateController;
    _controllee = controllee;
}
StateState2::~StateState2()
{
}
BaseState* StateState2::TryTransitWithoutEvent()
{
    _controllee->DoAction3();
    return nullptr;
}
const char* StateState2::GetStateName()
{
    return "State2";
}
BaseState* StateState2::GetParent()
{
    return nullptr;
}
int StateState2::GetStateID()
{
    return 1;
}